pkg/openstack/loadbalancer_test.go

```diff
@@ -1754,3 +1754,99 @@ func TestBuildBatchUpdateMemberOpts(t *testing.T) {
 		})
 	}
 }
+
+func TestLbaasV2_getNetworkID(t *testing.T) {
+	lbaas := LbaasV2{
+		LoadBalancer: LoadBalancer{
+			opts: LoadBalancerOpts{
+				LBClasses: map[string]*LBClass{
+					"lbclassKey": {
+						NetworkID: "lb-class-network-id-1234",
+					},
+				},
+				NetworkID: "default-lb-class-networkId",
+			},
+		},
+	}
+
+	type testArg struct {
+		service *corev1.Service
+	}
+	tests := []struct {
+		name    string
+		lbaas   LbaasV2
+		arg     testArg
+		want    string
+		wantErr string
+	}{
+		{
+			name:  "get network id from service annotation",
+			lbaas: LbaasV2{},
+			arg: testArg{
+				service: &corev1.Service{
+					ObjectMeta: v1.ObjectMeta{
+						Annotations: map[string]string{
+							ServiceAnnotationLoadBalancerNetworkID: "subnet-id",
+						},
+					},
+				},
+			},
+			want:    "subnet-id",
+			wantErr: "",
+		},
+		{
+			name:  "get network id from config class",
+			lbaas: lbaas,
+			arg: testArg{
+				service: &corev1.Service{
+					ObjectMeta: v1.ObjectMeta{
+						Annotations: map[string]string{
+							ServiceAnnotationLoadBalancerClass: "lbclassKey",
+						},
+					},
+				},
+			},
+			want:    "lb-class-network-id-1234",
+			wantErr: "",
+		},
+		{
+			name:  "get network id from default config",
+			lbaas: lbaas,
+			arg: testArg{
+				service: &corev1.Service{},
+			},
+			want:    "default-lb-class-networkId",
+			wantErr: "",
+		},
+		{
+			name:  "error when loadbalancer class not found",
+			lbaas: LbaasV2{},
+			arg: testArg{
+				service: &corev1.Service{
+					ObjectMeta: v1.ObjectMeta{
+						Annotations: map[string]string{
+							ServiceAnnotationLoadBalancerClass: "invalid-lb-class",
+						},
+					},
+				},
+			},
+			want:    "",
+			wantErr: "invalid loadbalancer class \"invalid-lb-class\"",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			lbaas := tt.lbaas
+
+			got, err := lbaas.getNetworkID(tt.arg.service, &serviceConfig{})
```

Comment:
just a nit: you don't need to use yet another variable

```suggestion
			got, err := tt.lbaas.getNetworkID(tt.arg.service, &serviceConfig{})
```

Reply:
I totally forgot, probably went back when I undo an error 
I had issues that I was fixing and undo some changes, so I think this sneaked in